### DISCUSSION TITLE
如何解决windows下，使用mingw平台链接时连接顺序导致符号找不到问题？

### DISCUSSION THREAD

**Discussion by v-dao** [xmake-io/xmake#1467] [Q&A] [answered]
> 注：如果是`问题报障`或者`特性请求`，请选择`报告缺陷`，`特性请求`模板，否则一律不回复。

### 描述讨论详情
请在这描述你需要讨论的话题详情。

```lua
-- xmake.lua
target("some_lib_used_socket")
    -- 这里用到了WSAStartup等ws2_32和wsock32的库函数。
    set_kind("static")
    includes("src/some_lib_used_socket")

target("app")
    set_kind("binary")
    includes("src/app")
    add_deps("some_lib_used_socket")
    add_links("wsock32", "ws2_32")
```

链接时打印：
C:\mingw64\bin\x86_64-w64-mingw32-g++ -o build\mingw\x86_64\release\app.exe -m64 -Lbuild\mingw\x86_64\release -lwsock32 -lws2_32 -lsome_lib_used_socket

报错：undefined reference to `__imp_WSAStartup'
当手动执行（把wsock32 ws2_32库放到后面）
C:\mingw64\bin\x86_64-w64-mingw32-g++ -o build\mingw\x86_64\release\app.exe -m64 -Lbuild\mingw\x86_64\release           
 -lsome_lib_used_socket -lwsock32 -lws2_32
就可以成功编译。
这个问题如果解决？

**Comment by waruqi** [comment, marked as answer]
用 add_syslinks

**Comment by v-dao** [comment]
哦，谢谢。当初看文档的时候没发现add_syslinks的作用，因为一直用cl编译器，没有遇到链接顺序的问题，直到用了gcc。
